tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py

```diff
@@ -0,0 +1,540 @@
+                if promise.get("version") == "latest":
+                    to_32 = "2"
+                elif promise.get("version"):
+                    to_32 = promise["version"]
+                elif from_32 == "0":
+                    to_32 = "2"
+                (to_64, to_32) = resolve_arch_conflicts(from_64, from_32, to_64, to_32)
+            else:
+                if promise.get("version"):
+                    version = ""
+                    if promise.get("version") == "latest":
+                        version = "2"
+                    else:
+                        version = promise["version"]
+                    if from_64 != "0" or (from_64 == "0" and from_32 == "0"):
+                        to_64 = version
+                    if from_32 != "0":
+                        to_32 = version
+                elif from_64 == "0" and from_32 == "0":
+                    to_64 = "2"
+
+        else:
+            # Todo: Errors when file doesn't match arch/version?
+            if promise["file_arch"] == "64_bit":
+                to_64 = promise["file_version"]
+            else:
+                to_32 = promise["file_version"]
+    else:
+        if promise.get("arch") == "64_bit":
+            if promise.get("version"):
+                if promise["version"] == from_64:
+                    to_64 = "0"
+            else:
+                to_64 = "0"
+        elif promise.get("arch") == "32_bit":
+            if promise.get("version"):
+                if promise["version"] == from_32:
+                    to_32 = "0"
+            else:
+                to_32 = "0"
+        else:
+            if promise.get("version"):
+                if promise["version"] == from_64:
+                    to_64 = "0"
+                if promise["version"] == from_32:
+                    to_32 = "0"
+            else:
+                to_64 = "0"
+                to_32 = "0"
+
+    return (to_64, to_32)
+
+
+# Calculate all possible transitions from from_state, to to_state.
+def calc_transitions(from_state, to_state):
+    from_64 = "0"
+    if states[from_state][("64_bit", "1")]:
+        from_64 = "1"
+    elif states[from_state][("64_bit", "2")]:
+        from_64 = "2"
+
+    from_32 = "0"
+    if states[from_state][("32_bit", "1")]:
+        from_32 = "1"
+    elif states[from_state][("32_bit", "2")]:
+        from_32 = "2"
+
+    to_64 = "0"
+    if states[to_state][("64_bit", "1")]:
+        to_64 = "1"
+    elif states[to_state][("64_bit", "2")]:
+        to_64 = "2"
+
+    to_32 = "0"
+    if states[to_state][("32_bit", "1")]:
+        to_32 = "1"
+    elif states[to_state][("32_bit", "2")]:
+        to_32 = "2"
+
+    promise_candidates = get_possible_promises()
+    valid_promises = []
+
+    for promise in promise_candidates:
+        result = simulate_promise(promise, from_64, from_32)
+        if result[0] == to_64 and result[1] == to_32:
+            valid_promises.append(promise)
+
+    return valid_promises
```

Comment:
hmm ... no, you want to iterate each of these twice, once for 'do' and once for 'count', and the iterable would be used up the first time round.  If you run into memory problems with running this script, it'd still be worth doing (just move the `calc_all_transitions()` call inside the 'count', 'do' loop) but I'm guessing memory isn't a problem for you.